qiskit/primitives/base/base_estimator.py

```diff
@@ -125,6 +126,16 @@ def __init__(
         self._observables = []
         self._parameters = []
         super().__init__(options)
+        # Raise warning if subclass doesn't override `run` method in anticipation of
+        # it becoming an `abstractmethod`
+        if self.run.__qualname__ == BaseEstimator.run.__qualname__:
+            warnings.warn(
+                "The `BaseEstimator.run` method being callable is deprecated as of Qiskit 0.46"
+                " and will be converted to an abstractmethod in Qiskit 1.0. Subclasses should"
+                " implement a `run` with the same signature and its own validation directly.",
```

Comment:
I don't think this migration path will work in practice for this release in particular there will be insufficient time for implementations of the interface to make this migration. For abstract interfaces we need to be more conservative about how we handle compatibility because we're not building an interface just for qiskit and end users, but also a shared interface that gets implemented by downstream libraries and used more broadly. We need to not just think about the upgrade strategy for Qiskit but also something that is depending on qiskit. If we go this route in practice it will mean that no external primitive implementations will be compatible with 1.0 which I don't think is a route we want to take. We really need to have support for both primitives implementing `run()` or `_run()` for 1.0 giving the time constraints.